lib/ts/fetch.ts

```diff
@@ -567,167 +566,203 @@ export async function onInvalidClaimResponse(response: ResponseWithBody) {
 
-    logDebugMessage("setIdRefreshToken: setting: " + idRefreshToken);
-    await setIDToCookie(idRefreshToken, AuthHttpRequest.config.sessionScope);
+    logDebugMessage("setTokenHeadersIfRequired: Adding st-auth-mode header: " + transferMethod);
+    clonedHeaders.set("st-auth-mode", transferMethod);
 
-    if (idRefreshToken === "remove" && status === "EXISTS") {
-        // we check for wasLoggedIn cause we don't want to fire an event
-        // unnecessarily on first app load or if the user tried
-        // to query an API that returned 401 while the user was not logged in...
-        if (statusCode === AuthHttpRequest.config.sessionExpiredStatusCode) {
-            logDebugMessage("setIdRefreshToken: firing UNAUTHORISED event");
-            AuthHttpRequest.config.onHandleEvent({
-                action: "UNAUTHORISED",
-                sessionExpiredOrRevoked: true,
-                userContext: {}
-            });
+    logDebugMessage("setTokenHeaders: adding existing tokens as header");
+
+    const token = await getTokenForHeaderAuth(addRefreshToken ? "refresh" : "access");
+    if (token) {
+        // the Headers class normalizes header names so we don't have to worry about casing
+        if (clonedHeaders.has("Authorization")) {
+            logDebugMessage("setTokenHeadersIfRequired: Authorization header defined by the user, not adding");
         } else {
-            logDebugMessage("setIdRefreshToken: firing SIGN_OUT event");
-            AuthHttpRequest.config.onHandleEvent({
-                action: "SIGN_OUT",
-                userContext: {}
-            });
+            if (token !== undefined) {
+                logDebugMessage("setTokenHeadersIfRequired: added authorization header");
+                clonedHeaders.set("Authorization", `Bearer ${token}`);
+            }
         }
+    } else {
+        logDebugMessage("setTokenHeadersIfRequired: token for header based auth not found");
     }
+}
 
-    if (idRefreshToken !== "remove" && status === "NOT_EXISTS") {
-        logDebugMessage("setIdRefreshToken: firing SESSION_CREATED event");
-        AuthHttpRequest.config.onHandleEvent({
-            action: "SESSION_CREATED",
-            userContext: {}
-        });
+async function saveTokensFromHeaders(response: Response) {
+    logDebugMessage("saveTokensFromHeaders: Saving updated tokens from the response headers");
+
+    const refreshToken = response.headers.get("st-refresh-token");
+    if (refreshToken) {
+        logDebugMessage("saveTokensFromHeaders: saving new refresh token");
+        const [value, expiry] = refreshToken.split(";");
+        await setToken("refresh", value, Number.parseInt(expiry));
+    }
+
+    const accessToken = response.headers.get("st-access-token");
+    if (accessToken) {
+        logDebugMessage("saveTokensFromHeaders: saving new access token");
+        const [value, expiry] = accessToken.split(";");
+        await setToken("access", value, Number.parseInt(expiry));
+    }
```

Comment:
Thinking about this, it would be easier if the backend just didn't set the expiry of these tokens and the frontend just set them to 100 years. That would then make the communication and implementation of non frontend SDK folks easier where they just have to set the value of `st-refresh-token` or `st-access-token` without the splitting stuff

Reply:
I'd prefer if the refresh token still had the expiration (although it's not critical), and we should keep them consistent.
Also, for cookie-based sessions, we'd still set this expiration on the BE so I think it makes sense to also have them in headers.

Reply:
hmm. Debatable, but ok. let's keep it the way it is cause it's implemented this way already.